scenario/build/generate.xml

```diff
@@ -42,4 +42,13 @@
             </argument>
         </arguments>
     </step>
+    <step name="run-baler" type="Magento\MagentoCloud\Step\Build\RunBaler" priority="1300">
+        <arguments>
+            <argument name="logger" xsi:type="object">Psr\Log\LoggerInterface</argument>
```

Comment:
Dependencies are autoloading automatically. You don't need to describe arguments, except you want to use a custom implementation of some interface or configure some static-type argument.

Reply:
Please ignore my comment above, I see that arguments are required in your case.